sound/soc/sof/ipc.c

```diff
@@ -393,7 +393,7 @@ EXPORT_SYMBOL(sof_ipc_tx_message_no_pm);
 void snd_sof_ipc_get_reply(struct snd_sof_dev *sdev)
 {
 	struct snd_sof_ipc_msg *msg = sdev->msg;
-	struct sof_ipc_reply reply;
+	struct sof_ipc_reply *reply = msg->reply_data;
```

Comment:
@ujfalusi  I have thought about this one before and I think one of the reasons to use a local variable and check for errors before copying is control put operations. The message and the reply point to the same data in this case isn't it so you could end up losing the cached data in case of an ipc error. I'm not sure if it's good or bad but just something to think about

Reply:
@ranj063, something is not right for sure and it is around the reply handling with this PR...

Before this patch this is what we did:
read the `reply` to a local struct from the mailbox.
if reply.error < 0, memcpy the `reply` to the msg->reply_data
if reply.error == 0, read the whole reply to msg->reply_data from the mailbox

with this patch:
read `reply` to msg->reply_data from the mailbox
if reply.error < 0, no need to copy, we already have it
if reply.error == 0 and the reply_size is more than the size of the generic reply, read the whole message again to msg->reply_data from the mailbox. otherwise no need to re-read it, we already have the reply.

Looks like alone this patch works, I need to figure out where I have messed up things.